### PR TITLE
Grant Application: Roloi Multi-Sig Vault

### PR DESCRIPTION
**Pull request by RoloiMoney**
**Grant Application: Roloi Multi-Sig Vault**

Roloi Multi-Sig Vault is a generic multi-key wallet that empowers the security of funds.

### CODE REVIEW

**Comment by TtomaS7** [comment]
Thank you for submitting your application. Our curator will promptly review it and provide any feedback or inquiries he may have. We appreciate your interest!

File: applications/roloi-multisig-vault.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+- **Payment Address:** `1dz667uQX199rHyj6tizmL6EJe4AZxjH1RhnyrT1QuQ4pfS`
+
+## Project Overview :page_facing_up:
+### Overview
+
+Multi-Sig Vault Canary Dapp. 
+
+### Project Details
+
+**Roloi Multi-Sig Vault** is a generic multi-key wallet that empowers the security of funds. The DApp allows **sharing a vault between multiple users** to enable collective asset management.
```
Comment (mike1729) [inline applications/roloi-multisig-vault.md:14]: Could you compare your solution to the existing native multisig? What are the advantages and what are disadvantages? In what scenario this multisig is batter than the native one?

File: applications/roloi-multisig-vault.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+- **Team Name:** NeoPower Digital
+- **Project Category:** Canary DApp
+- **Payment Address:** `1dz667uQX199rHyj6tizmL6EJe4AZxjH1RhnyrT1QuQ4pfS`
+
+## Project Overview :page_facing_up:
+### Overview
+
+Multi-Sig Vault Canary Dapp. 
+
+### Project Details
```
Comment (mike1729) [inline applications/roloi-multisig-vault.md:12]: Could you describe some cryptographic details of your solution? For example, how is the address of a vault generated?

**Review by mike1729 — CHANGES_REQUESTED** [review]

**Review by 0xMarkian — CHANGES_REQUESTED** [review]
Hey guys! thanks for the application!
I would like to agree with @mike1729. I am not sure creating non native multisig is really needed at least at this point of time.

**Comment by SotaWatanabe** [comment]
Thanks for your application. Agreed with Markian and Mike. I prefer to have a secure native custody solution like MPC or Account Abstraction rather than ink! smart contract vault.

**Comment by RoloiMoney** [comment]
@mike1729 @SotaWatanabe @0xMarkian 
We answered your questions, we’d appreciate your feedback if you think there is a way to improve or change our proposal. Thanks!

**Comment by DoubleOTheven** [comment]
I like this proposal a lot and think it can be beneficial to many organizations using ink! EVM has many vaults including [Gnosis Safe](https://safe.global/), which I use personally. Many many NFT projects use it. We could use something like this that allows users to customize their vaults in ways that you cannot do so in the native way.

What you already pointed out is good:

```
    * It doesn’t allow updating the owners or the required signers number
    * Polkadot.js is frustrating for users to create their own multi-sigs
    * It demands a deposit for each new call to be executed
```

I suggest that you identify more ways people can customize their vault and think of it like Legos for Multi-Sig. You could also build a nice UI/UX using something like [useink](https://github.com/paritytech/useink). Take a look at the features in Gnosis and others like it for inspiration.

What I also like about this proposal is that it would provide a foundational code base that others can fork and use to innovate. You don't get that with a permission-ed runtime feature. In addition, it would be another good example of ink! in production.

**Comment by DoubleOTheven** [comment]
There is another [multi-sig proposal here](https://github.com/use-inkubator/Ecosystem-Grants/pull/14/files). I will say that the other team has the advantage of previously launching a Gnosis-like dApp for EVM.

For this proposal, you should think of a way you can differentiate yourself from the other one or perhaps join forces.

File: applications/roloi-multisig-vault.md (hunk 2 above)
Comment (mike1729) [reply applications/roloi-multisig-vault.md:12]: Then users will have to pay for txs they want to cosign instead of deposits in native multisig, right?

**Review by mike1729 — CHANGES_REQUESTED** [review]
In general, I believe that multisig functionality implemented as a smart contract is very important as this way it will be easier to integrate with other smart contracts than with native multisig. What is lacking in my opinion is some details on the design of the underlying cryptographic logic as currently there is limited support for different elliptic curves and in general the env of SC is more constrained than pallets so what is possible in native may be too expensive in contracts.

File: applications/roloi-multisig-vault.md (hunk 2 above)
Comment (RoloiMoney) [reply applications/roloi-multisig-vault.md:12]: Exactly. On this version, the signer that executes the transaction will have to pay it (after the transaction is properly approved).

**Comment by RoloiMoney** [comment]
Thanks for the feedback! We really appreciate it. Here is some more context about our team:
- We are currently building the **first payment automation platform in Astar**, named **[Roloi](https://www.roloi.money/)**, based on ink! smart contracts. The **multi-sig vault** will be an **essential feature** of our platform.
- Roloi is the **first ink! project of LATAM**. Last year, we received a **grant from the Web3 Foundation** to migrate our contracts **from CosmWasm to ink!** and published a [Medium article](https://medium.com/@RoloiMoney/cosmwasm-to-ink-beyond-the-cosmos-e4920604f9cb) about our experience.
- As active members of the LATAM community, we have organized **[ink! workshops and talks](https://www.youtube.com/@NeoPower-Digital/videos)**, and participated as **judges** in the **Polkadot Hackathon LATAM 2022**. 
- In July, we will present the **first ink! course in LATAM** as part of a Polkadot Hub initiative. 
- We have been working together with Michi Muller to t**ranslate the ink! official documentation to Spanish**. 
- We are constantly **building relationships** with other teams from the ecosystem, and we think building public goods is a great way to accelerate the development of our product while benefiting other teams with open-source code.
- In a **future version**, this UI can allow the creation of native multi-sigs interacting with the pallet too. This way the user can choose which multisig approach to use.

**Comment by mike1729** [comment]
@RoloiMoney I asked about a design of the logic of your multisig and you provided me with context about your team. Can you please answer my original question?

**Comment by RoloiMoney** [comment]
> @RoloiMoney I asked about a design of the logic of your multisig and you provided me with context about your team. Can you please answer my original question?

Hi, @mike1729. Sorry for that, we answered the comments of Sam.

About cryptographic details: 
- We will instantiate multi-sig vault smart contracts and use the provided standard addresses.
- The signatures required to confirm a multi-sig transaction will follow the contracts pallet standard too. We won't provide any extra cryptographic logic.
- The process will include: 
  - One extrinsic to create a transaction to be confirmed.
  - N extrinsics to confirm the transaction.
  - One extrinsic to execute the transaction after reaching the required number of confirmations. 

We are aware of the limitations of a smart contract vs runtime code. We believe that with this solution we can explore and test multiple features of known multi-sigs in an easier and more flexible way.
We are open to connecting the UI with the native multi-sig pallet in a future iteration, since this would allow users to decide on which approach to use based on the trade-offs of each option.

Thank you for your questions and feedback. We’d appreciate any comment or suggestion to improve the solution.

**Comment by TtomaS7** [comment]
@RoloiMoney 
I would like to start off by expressing our deepest appreciation for the time, effort, and innovation you have put into your application for our Ecosystem Grant. 

Regrettably, after a careful and thorough review, we are unable to offer you the grant at this time. Also, we have recently approved another multisig solution which overlaps considerably with your proposal. This has been a critical factor in our decision-making process.
In addition, as per our process and guidelines, an application should ideally receive approval within 4 weeks of its submission. Unfortunately, the requisite approval has not been secured in your case within this timeline.

Please do not take this decision as a reflection of the quality of your work or the value of your ideas. We truly recognize and appreciate your contributions, and we believe your project has a potential to provide significant value to the community.

We encourage you to re-apply in the future, especially if you have any new, innovative ideas or approaches.
Once again, we thank you for your interest in our Ecosystem Grant and your understanding in this matter.

**Pull request closed without merging**